virtool/samples/data.py

```diff
@@ -158,29 +158,32 @@ async def find(
         )
 
     async def get(self, sample_id: str) -> Sample:
-        document = await self._db.samples.find_one({"_id": sample_id})
-
-        if not document:
-            raise ResourceNotFoundError
+        async for document in self._db.samples.aggregate(
+            [
+                {"$match": {"_id": sample_id}},
+                *lookup_nested_user_by_id(local_field="user.id")
+            ]
+        ):
+            if not document:
+                raise ResourceNotFoundError
```

Comment:
As written I don't think this can ever occur. If there is no document then it will never enter the loop